src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -87,26 +120,10 @@ export default class EthereumBlockchainService implements BlockchainService {
       nonce: baseNonce,
     };
 
-    const { overrideGasConfig } = config.blockchain.connectors.ethereum;
-    if (config.blockchain.connectors.ethereum.overrideGasConfig) {
-      txData.gasPrice = BigNumber.from(config.blockchain.connectors.ethereum.gasPrice);
-      logger.debug('Overriding Gas price: ' + txData.gasPrice.toString());
-
-      txData.gasLimit = BigNumber.from(config.blockchain.connectors.ethereum.gasLimit);
-      logger.debug('Overriding Gas limit: ' + txData.gasLimit.toString());
-    }
-
-    let retryTimes = 3;
-    while (retryTimes > 0) {
+    let attemptNum = 0;
+    while (attemptNum < MAX_RETRIES) {
```

Comment:
Not totally clear to me what the right call here is.  I don't really have enough ethereum experience to say whether or not there are other failure modes that might be recoverable with a retry.

Reply:
The fact that even if we fail the whole batch we would still just start a new batch shortly thereafter definitely complicates the error handling story here.  I'm starting to think more and more that we should be handling scheduling within a long-running CAS instead of starting a new instance every batch

Reply:
> The fact that even if we fail the whole batch we would still just start a new batch shortly thereafter definitely complicates the error handling story here. I'm starting to think more and more that we should be handling scheduling within a long-running CAS instead of starting a new instance every batch

Maybe we made an optimistic optimization here without knowing what it would look like ultimately. Though I'll add that if it is a problem of how we are running the process we can go down the route of just a better way to do that. As an example, add a check to the docker image to see if we already have an ECS task for the anchor service running and if so don't spin up another one.

So we can evaluate where to place the solution based on whatever is easiest + simplifies this.

Reply:
> Do we risk losing eth here?

Not really the only thing that could happen is that our tx doesn't get mined because the `gasPrice` is to low.

The `gasLimit` should always be static for the type of transaction we are making.

Reply:
> > Do we risk losing eth here?
> 
> Not really the only thing that could happen is that our tx doesn't get mined because the `gasPrice` is to low.
> 
> The `gasLimit` should always be static for the type of transaction we are making.

@stbrody this is kind of the point that I want to be more clear. Since this is not a smart contract tx it really is fine to do it the way we are because there are really only 2 failure modes--one where increasing gas price helps and the other where increasing gas price doesnt do anything. The issue I see in the way this code is written is that the knowledge of this is implicit.

Maybe an easy way to refactor this is

```
while(retrying) {
  try sendSimpleTx & break
  catch timeout & increase gasPrice
}
```

But up to you if it's that important rn.

Reply:
I don't actually follow what your suggestion here is?  That looks basically like how this code is already structured to me.